### DISCUSSION TITLE
Keine Daten mehr vom Wechselrichter...

### DISCUSSION THREAD

**Discussion by frankhirsch** [tbnobody/OpenDTU#940] [Q&A] [answered]
Hallo, nachdem ich mir die OpenDTU erstmals vor ein paar Monaten installiert hatte wurden die Daten zunächst korrekt in vernünftigen Zeitintervallen aus dem HM-700 ausgelesen. In den letzten 3 Wochen hatte ich dann aber festgestellt dass die Daten _immer weniger häufiger_ im WebUI aktualisiert wurden. Trotz Sonnenschein und grün blinkender LED am Wechselrichter wurden die Daten nur noch sehr sporadisch aktualisiert (teilweise ≥ 2 Stunden) - bis dann irgendwann gar nichts mehr ging. Ausgehend von verschiedenen Hinweisen ging ich von einem defekten Funkmodul aus, aber trotz Tausch und letztlich der Neuinstallation der OpenDTU bekomme ich nun leider gar keine Daten vom WR mehr.

Ich hatte zunächst den ESP32 mit einer selbst kompilierten Version (VisualStudio/Platform.io unter MacOS) geflasht und dann auch die Installation der Binärdistribution über WebUI) probiert.

Das Funkmodul wird im WebUI immer erkannt (siehe: System Information), jedoch kommt neuerdings scheinbar keine Verbindung mehr zum WR zustande (siehe: Konsole). Die zuvor funktionierende Konfiguration hatte ich mir zur Sicherheit extra noch als JSON exportiert...

Hätte jemand eine Idee oder Tips ob ich da etwas falsch gemacht habe - oder ggf. auch einen Tip wie ich über Konsole oder VS Prüfen/Debuggen könnte ob mein WR überhaupt noch sendet/antwortet?

<img width="769" alt="01" src="https://github.com/tbnobody/OpenDTU/assets/1833680/9f4da3c9-702f-4658-acb8-8806d1ce0fb9">
<img width="770" alt="02" src="https://github.com/tbnobody/OpenDTU/assets/1833680/fab8a45e-6ef5-4683-8962-077809993175">

**Comment by frankhirsch** [comment]
Eventuell kann ja auch jemand sehen ob ich trotz allem noch einen Verkabelungsfehler habe... da die Bezeichnungen bei meiner Antenne nicht 100% identisch zu den Definitionen in der Default pin_mapping.json sind könnte das im Zweifel der Fall sein...
<img width="1259" alt="03" src="https://github.com/tbnobody/OpenDTU/assets/1833680/e904efb1-ca77-40e8-89cb-47b648f1b6bf">

**Reply by Popelbittler** [reply, marked as answer]
Hallo frankhirsch,
ich hatte ein ähnliches Problem. Die Lösung: Austausch des Funkmoduls gegen eines mit geschirmten Hochfrequenzteil.
Den Tipp habe ich in iregendeinem Forum aufgeschnappt. Was habe ich mich schon geärgert und rumexperimentiert.
Es gibt vom Hersteller ebyte, Verkauf durch das große Internetkaufhaus, pinkompatible Module. 
Die funktionieren einwandfrei. Es ist kein Pinning aufgedruckt. Das Pinning ist aber kompatibel.
Ich bekomme auf die Schnelle kein Foto eingebunden. Suche nach:
EBYTE 2,4 GHz Wireless Modul nRF24L01 + PA + LNA / E01-ML01DP5
Das Modul hat eine Blechkappe mit der Aufschrift u.a.  E01 ML01DP5.
Das Modul das Du verwendest taugt definitiv nichts!!! Nachdem ich mit dem anderen Modul sehr gute Erfahrungen gemacht habe, habe ich das ungeschirmte Modul in ein Blechgehäuse verpackt und zugelötet. - Hat nichts gebracht.
Viel Erfolg, wüede mich über Feedback freuen.

**Reply by Popelbittler** [reply]
![E01-ebyte ](https://github.com/tbnobody/OpenDTU/assets/128311391/ba07d744-9050-4550-ad58-ea16b0481ad4)
Ich hoffe das Bild ist sichtbar

**Reply by Popelbittler** [reply]
Hallo frankhirsch,
wohnst Du zufällig in der Nähe von Karlsruhe?  Ich könnte Dir so ein ebyte Modul gerne mal ausleihen.
Ich versorge hier die Nachbarschaft mit openDTU Hardware, damit sie ihren Ertrag ablesen können.
Weil die Nachbarn wenig bis null IT-affin sind konfiguriere ich die Parameter. Deshalb kenne ich die Seriennummern von zwei Invertern und meine natürlich.
Ich empfange hier testweise alle drei Inverter, zwei davon Luftlinie ca. je 70 m ohne direkte Sichtverbindung.
Bei der Abfrage meines eigenen Hoymiles kommen die aktualisierten Werte alle 5 sec. wie ein Uhrwerk.
Das alles erst nach Wechsel des Funkmoduls. :-)) und nach langem Experimentieren mit Antennenausrichtung, verschiedenen Aufstellorten usw.

**Comment by Omega13x** [comment]
Hast du dir in den letzten drei Wochen irgend ein neues Wlan Gerät oder irgend ein anderes Gerät zugelegt das im 2,4Ghz Bereich sendet und in der Nähe der DTU bzw zw DTU und Wechselrichter platziert? Ich hatte vor etwa fünf Wochen das gleiche Problem und habe ewig gesucht. Am Ende war es mein Mähroboter den ich wieder in Betrieb genommen hatte. Ich habe dann die DTU an einem anderen Ort montiert und schon ging das Ganze wieder.

**Reply by frankhirsch** [reply]
Interessanter Ansatz! Klar, es funkt ja ziemlich viel auf 2,4 GHz… auch mein Sileno denke ich den ich in dem Zeitraum aus dem Winterschlaf geholt habe. Werde ich auf jeden Fall mal alles abschalten und testen!

**Comment by frankhirsch** [comment]
Danke Leute - ich habe es in den Griff bekommen!

Am Ende war es eine Mischung aus 1. nicht mehr nachvollziehbaren technischen Umständen (gestörtes Funksignal, defektes Funkmodul… dass die alte funktionierende Installation zunehmend schlechter reagierte), 2. einer möglicherweise schlechten JSON Konfiguration im aktuellen Standard (eth war nicht FALSE) und 3. meiner Unachtsamkeit (WR-Seriennummer war fehlerhaft)… Nachdem ich mir dann nochmal alles in Ruhe angeschaut habe und das Debugging im Visualstudio vorgenommen habe konnte ich das auch rausfinden und beheben!

**Comment by Rafael3366** [comment]
Ich habe mir auch einen Open DTU mit dem ESP 32 V1 gebaut.

Aber ich bekomme auch keine Daten vom Wechselrichter.

Ich weiẞ gerade nicht ob es am der Verdrahtung oder an diesem Mapping der Ausgänge. 
Kann mir jemand seine .json Datei senden damit ich es probieren kann.
Ein neues wlan Modul nRF24L01+ habe ich mir auch schon bestellt 
Das sind meine Einstellungen:
Sehe ich hier das meine Einstellung richtig sind und das meine WLAN Modul sendet ? 

![image](https://github.com/user-attachments/assets/2d54c8c4-8c7b-47da-baf8-7533f67f185c)

So sehen meine Pin Einstellungen aus sind diese für diese Platinen richtig 

![image](https://github.com/user-attachments/assets/ed067861-0097-412e-9722-3279e855e25a)

**Reply by SW-Niko** [reply]
Hallo @Rafael3366 ,
die Config muss zu deiner Verdrahtung passen. Diese beiden Dinge gehören zusammen.
Es macht ja auch keinen Sinn wenn ich dir meinen Stadtplan sende ohne zu wissen in welcher Stadt du bist. 😉 
Mit anderen Worten... ohne Verdrahtungsplan kann dir vermutlich keiner helfen.

**Reply by Rafael3366** [reply]
Ich danke dir für diese Information. Mir ist es schon bewusst, dass es da sehr viele Möglichkeiten gibt aber da ich hier kompletter Neuling bin, ob ich eigentlich die richtigen Fragen stellen kann ich weiß nur, dass mein Gerät mit diesen Einstellungen nicht funktioniert.

**Reply by SW-Niko** [reply]
Moment.... Mein Fehler. 😞 
Laut deinem Screenshot ist die Verbindung zum nRF24 in Ordnung.
Du hast schon einen HM-xxx und keinen HMS-xxx? 🤔

**Reply by Rafael3366** [reply]
Ich habe die  hm-1500

**Reply by SW-Niko** [reply]
Also...
Du hast einen HM-1500, und das richtige Funkmodul, und Verbindung zum Funkmodul.

Was steht in der Funkstatistik und wie weit ist der Inverter vom Funkmodul entfernt?
